pkg/pillar/cmd/zedmanager/handleresolve.go

```diff
@@ -0,0 +1,183 @@
+// Copyright (c) 2017-2018 Zededa, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+package zedmanager
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/lf-edge/eve/pkg/pillar/types"
+	log "github.com/sirupsen/logrus"
+)
+
+// MaybeAddResolveConfig will publish the resolve config for
+// container images for which resolution of tags to sha requires
+func MaybeAddResolveConfig(ctx *zedmanagerContext, ss *types.StorageStatus) {
+
+	log.Infof("MaybeAddResolveConfig for %s", ss.ImageID)
+	resolveConfig := types.ResolveConfig{
+		DatastoreID: ss.DatastoreID,
+		Name:        ss.Name,
+		AllowNonFreePort: types.AllowNonFreePort(*ctx.globalConfig,
+			types.AppImgObj),
+		Counter: ss.PurgeCounter,
+	}
+	publishResolveConfig(ctx, &resolveConfig)
+	log.Infof("MaybeAddResolveConfig for %s Done", ss.ImageID)
+}
+
+func publishResolveConfig(ctx *zedmanagerContext,
+	config *types.ResolveConfig) {
+
+	key := config.Key()
+	log.Debugf("publishResolveConfig(%s)\n", key)
+	pub := ctx.pubAppImgResolveConfig
+	pub.Publish(key, *config)
+	log.Debugf("publishResolveConfig(%s) Done\n", key)
+}
+
+func unpublishResolveConfig(ctx *zedmanagerContext,
+	config *types.ResolveConfig) {
+
+	key := config.Key()
+	log.Debugf("unpublishResolveConfig(%s)\n", key)
+	pub := ctx.pubAppImgResolveConfig
+	pub.Unpublish(key)
+	log.Debugf("unpublishResolveConfig(%s) Done\n", key)
+}
+
+func lookupResolveConfig(ctx *zedmanagerContext,
+	key string) *types.ResolveConfig {
+
+	log.Infof("lookupResolveConfig(%s)\n", key)
+	pub := ctx.pubAppImgResolveConfig
+	c, _ := pub.Get(key)
+	if c == nil {
+		log.Infof("lookupResolveConfig(%s) not found\n", key)
+		return nil
+	}
+	config := c.(types.ResolveConfig)
+	log.Infof("lookupResolveConfig(%s) Done\n", key)
+	return &config
+}
+
+func lookupResolveStatus(ctx *zedmanagerContext,
+	key string) *types.ResolveStatus {
+
+	log.Infof("lookupResolveStatus(%s)\n", key)
+	sub := ctx.subAppImgResolveStatus
+	c, _ := sub.Get(key)
+	if c == nil {
+		log.Infof("lookupResolveStatus(%s) not found\n", key)
+		return nil
+	}
+	status := c.(types.ResolveStatus)
+	log.Infof("lookupResolveStatus(%s) Done\n", key)
+	return &status
+}
+
+func handleResolveStatusDelete(ctxArg interface{}, key string,
+	statusArg interface{}) {
+
+	log.Infof("handleResolveStatusDelete for %s\n", key)
+	ctx := ctxArg.(*zedmanagerContext)
+	status := statusArg.(types.ResolveStatus)
+	config := lookupResolveConfig(ctx, status.Key())
+	if config != nil {
+		log.Infof("handleResolveStatusDelete delete config for %s\n",
+			key)
+		unpublishResolveConfig(ctx, config)
+	}
```

Comment:
This is backwards, and I hope it is never executed in your tests. Would make sense to check the logs.

The deletion of the config will result in downloader deleting the status. But not the other way around.
Thus I think handleResolveStatusDelete can log but otherwise do nothing.

Reply:
Updated.